atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/proofOfPresentation/EdgePresentationRequest.kt

```diff
@@ -0,0 +1,155 @@
+        if (thid != other.thid) return false
+        if (from != other.from) return false
+        if (to != other.to) return false
+        return type == other.type
+    }
+
+    /**
+     * Calculates the hash code for the [RequestPresentation] object.
+     *
+     * The hash code is calculated based on the following properties of the [RequestPresentation] object:
+     * - [id]
+     * - [body]
+     * - [attachments]
+     * - [thid]
+     * - [from]
+     * - [to]
+     * - [type]
+     *
+     * @return The hash code value for the [RequestPresentation] object.
+     */
+    override fun hashCode(): Int {
+        var result = id.hashCode()
+        result = 31 * result + body.hashCode()
+        result = 31 * result + attachments.contentHashCode()
+        result = 31 * result + (thid?.hashCode() ?: 0)
+        result = 31 * result + from.hashCode()
+        result = 31 * result + to.hashCode()
+        result = 31 * result + type.hashCode()
+        return result
+    }
+
+    companion object {
+        /**
+         * Converts a given [Message] object to a [RequestPresentation] object.
+         *
+         * @param fromMessage The [Message] object to convert.
+         * @return The converted [RequestPresentation] object.
+         * @throws PrismAgentError.InvalidMessageType if the [Message] object does not represent the correct protocol
+         *         or if it is missing the "from" and "to" fields.
+         */
+        @JvmStatic
+        @Throws(PrismAgentError.InvalidMessageType::class)
+        fun fromMessage(fromMessage: Message): EdgePresentationRequest {
+            if (fromMessage.piuri == ProtocolType.DidcommPresentationDefinition.value &&
+                fromMessage.from != null &&
+                fromMessage.to != null
+            ) {
+                return EdgePresentationRequest(
+                    id = fromMessage.id,
+                    body = Json.decodeFromString(fromMessage.body),
+                    attachments = fromMessage.attachments,
+                    thid = fromMessage.thid,
+                    from = fromMessage.from,
+                    to = fromMessage.to
+                )
+            } else {
+                throw PrismAgentError.InvalidMessageType(
+                    type = fromMessage.piuri,
+                    shouldBe = ProtocolType.DidcommPresentationDefinition.value
+                )
+            }
+        }
+    }
+
+    @Serializable
+    data class Body @JvmOverloads constructor(
+        val attachmentFormats: Array<CredentialFormat>? = null
+    ) {
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as Body
+
+            return attachmentFormats.contentEquals(other.attachmentFormats)
+        }
+
+        override fun hashCode(): Int {
+            return attachmentFormats.contentHashCode()
+        }
+    }
+}
```

Comment:
Uhn, I know what you're trying to do here. But we cannot have a message just for Edge Agents presentations, we need to be interoperable.

So it should use message protocols already defined for didcomm, example Aries uses a version for didcomm V1, so we should use the V3.0 messages protocol for didcomm v2 and not create a new protocol.

https://github.com/decentralized-identity/waci-didcomm/blob/main/present_proof/present-proof-v3.md

Reply:
After a conversation about this, we agreed to re-use RequestPresentation instead of creating a new Message type. I deleted EdgeRequestPresentation.